backend/prompt_studio/prompt_studio_v2/models.py

```diff
@@ -0,0 +1,127 @@
+import uuid
+
+from account_v2.models import User
+from django.db import models
+from prompt_studio.prompt_profile_manager_v2.models import ProfileManager
+from prompt_studio.prompt_studio_core_v2.models import CustomTool
+from utils.models.base_model import BaseModel
+
+
+class ToolStudioPrompt(BaseModel):
+    """Model class while store Prompt data for Custom Tool Studio.
+
+    It has Many to one relation with CustomTool for ToolStudio.
+    """
+
+    class EnforceType(models.TextChoices):
+        TEXT = "Text", "Response sent as Text"
+        NUMBER = "number", "Response sent as number"
+        EMAIL = "email", "Response sent as email"
+        DATE = "date", "Response sent as date"
+        BOOLEAN = "boolean", "Response sent as boolean"
+        JSON = "json", "Response sent as json"
+
+    class PromptType(models.TextChoices):
+        PROMPT = "PROMPT", "Response sent as Text"
+        NOTES = "NOTES", "Response sent as float"
+
+    class Mode(models.TextChoices):
+        DEFAULT = "Default", "Default choice for output"
+
+    prompt_id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
+    prompt_key = models.TextField(
+        blank=False,
+        db_comment="Field to store the prompt key",
+    )
+    enforce_type = models.TextField(
+        blank=True,
+        db_comment="Field to store the type in \
+            which the response to be returned.",
+        choices=EnforceType.choices,
+        default=EnforceType.TEXT,
+    )
+    prompt = models.TextField(
+        blank=True, db_comment="Field to store the prompt", unique=False
+    )
+    tool_id = models.ForeignKey(
+        CustomTool,
+        on_delete=models.SET_NULL,
+        related_name="tool_studio_prompts",
+        null=True,
+        blank=True,
+    )
```

Comment:
@muhammad-ali-e Shouldn't this be `on_delete=models.PROTECT`?
The prompts should not exist if the parent custom tool is deleted, right?

Reply:
@hari-kuriakose I think , It shouldn't needed When itool is deleted . So It shoudl CASCADE
@harini-venkataraman @chandrasekharan-zipstack Can you confirm here?

Reply:
@hari-kuriakose @muhammad-ali-e I think it should be `CASCADE` here since we should delete the individual prompts when the tool is deleted. On the other hand if we used `PROTECT`, we won't be able to delete the prompt studio project until all the prompts within it are deleted which is not what we want.
Referred [this SO answer](https://stackoverflow.com/a/38389488/12864260)

Reply:
Updated